File: ADA/sorting/bubble_sort.go
package main
import "fmt"

func main() {
	arr := [] int {3, 4, 1, 544, 2}
	fmt.Println(BubbleSort(arr))
} 

// will take an aray as parameter and return an array
func BubbleSort(arr[] int) [] int {
	for i := 0; i < len(arr) - 1; i++ {
		for j := 0; j < len(arr) - i - 1; j++ {
			if(arr[j] > arr[j+1]) {
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}
	return arr
}

File: ADA/sorting/bubble_sort.cpp
#include <iostream>
using namespace std;

void bubbleSort (int arr [], int size) {
    int i, j;
    
    for (int i = 0; i < size; i++) {
        // after each iteration of i, we will get max value at end
        for (int j = 0; j < size - i - 1; j++) { 
            if (arr[j] > arr[j+1]) {
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }

    for(int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }
    
    cout << endl;
    cout << "Complexity : O(n^2)" << endl;

}

int main() {
    
    cout << "Achintya" << " " << "A2305220450" << endl;
    cout << "Enter size of array" << endl;
    int size;
    cin >> size;
    int arr[size];

    cout << "Enter your array" << endl;
    for (int i = 0; i < size; i++) {
        cin >> arr[i];
    }
    
    size = sizeof(arr)/sizeof(int);
    bubbleSort(arr, size);
}